src/mgmt/cli_mgmt.c

```diff
@@ -583,41 +578,37 @@ int dc_mgmt_net_cfg(const char *name)
 
 	/* Allow client env overrides for these three */
 	d_agetenv_str(&crt_timeout, "CRT_TIMEOUT");
-	if (!crt_timeout) {
-		rc = asprintf(&crt_timeout, "%d", info->crt_timeout);
-		if (rc < 0) {
-			crt_timeout = NULL;
-			D_GOTO(cleanup, rc = -DER_NOMEM);
-		}
-		D_INFO("setenv CRT_TIMEOUT=%s\n", crt_timeout);
-		rc = d_setenv("CRT_TIMEOUT", crt_timeout, 1);
-		if (rc != 0)
-			D_GOTO(cleanup, rc = d_errno2der(errno));
-	} else {
+	if (!crt_timeout)
+		crt_info->cio_crt_timeout = info->crt_timeout;
+	else
 		D_DEBUG(DB_MGMT, "Using client provided CRT_TIMEOUT: %s\n", crt_timeout);
-	}
-
-	/* client-provided iface/domain were already taken into account by agent */
-	/* TODO: These should be set in crt_init_options_t instead of env */
-	rc = d_setenv("D_INTERFACE", info->interface, 1);
-	if (rc != 0)
-		D_GOTO(cleanup, rc = d_errno2der(errno));
-
-	rc = d_setenv("D_DOMAIN", info->domain, 1);
-	if (rc != 0)
-		D_GOTO(cleanup, rc = d_errno2der(errno));
 
 	sprintf(buf, "%d", info->provider_idx);
 	rc = d_setenv("CRT_SECONDARY_PROVIDER", buf, 1);
 	if (rc != 0)
 		D_GOTO(cleanup, rc = d_errno2der(errno));
```

Comment:
@frostedcmos Do we need to add a field for additional providers to `crt_init_options_t`?

Reply:
yea i think it would be great to avoid using internally any env variables like this.

Reply:
> @frostedcmos Do we need to add a field for additional providers to `crt_init_options_t`?

No, for secondary providers they are all specified as a coma separated list in existing cio_provider/cio_interface/cio_domain/cio_port etc... strings. This particular field is a bool flag to indicate to the client side of cart whether we are running on a primary or a secondary cluster. 

We should indeed add a corresponding bool cio_is_secondary; to crt_init_options_t at some point. Good time to do it would probably be when we enable multiprovider support.